V2/gameFuseGameRound.js

```diff
@@ -0,0 +1,296 @@
+                GameFuse.Log("GameFuseGameRound update Success");
+
+                // update the game round object with the response data returned from the API
+                let updatedRoundData = response.data;
+                Object.assign(this, {
+                    id: updatedRoundData.id,
+                    gameID: updatedRoundData.game_id,
+                    gameType: updatedRoundData.game_type,
+                    startTime: updatedRoundData.start_time && new Date(updatedRoundData.start_time),
+                    endTime: updatedRoundData.end_time && new Date(updatedRoundData.end_time),
+                    score: updatedRoundData.score,
+                    place: updatedRoundData.place,
+                    metadata: updatedRoundData.metadata,
+                    multiplayerGameRoundID: updatedRoundData.multiplayer_game_round_id,
+                    rankings: updatedRoundData.multiplayer_game_round_id == null ? [] : this.constructor.buildRankings(updatedRoundData.rankings)
+                })
+            }
+
+            GameFuseUtilities.HandleCallback(
+                response,
+                responseOk ? 'game round has been created successfully' : response.data, // message from the api
+                callback,
+                !!responseOk
+            )
+        } catch (error) {
+            console.log(error);
+            GameFuseUtilities.HandleCallback(typeof response !== 'undefined' ? response : undefined, error.message, callback, false)
+        }
+    }
+
+
+    async destroy(callback = undefined) {
+        try {
+            GameFuse.Log('Destroying game round');
+
+            const url = `${GameFuse.getBaseURL()}/game_rounds/${this.getID()}`;
+            let currentUser = GameFuseUser.CurrentUser;
+
+            const response = await GameFuseUtilities.processRequest(url, {
+                method: 'DELETE',
+                headers: {
+                    'Content-Type': 'application/json',
+                    'authentication-token': currentUser.getAuthenticationToken()
+                }
+            });
+
+            const responseOk = await GameFuseUtilities.requestIsOk(response);
+            if (responseOk) {
+                GameFuse.Log('GameFuseGameRound destroy game round success');
+
+                // remove the object from the state.
+                currentUser.gameRounds = currentUser.gameRounds.filter(gameRound => gameRound.getID() !== this.getID());
+            }
+
+            GameFuseUtilities.HandleCallback(
+                response,
+                responseOk ? 'Game Round destroyed successfully' : response.data, // message from the api
+                callback,
+                !!responseOk
+            )
+        } catch (error) {
+            console.log(error);
+            GameFuseUtilities.HandleCallback(typeof response !== 'undefined' ? response : undefined, error.message, callback, false)
+        }
+    }
+
+    static buildRankings(apiData) {
```

Comment:
what is this function used for? Is it sorting multiplayer round scores?

Reply:
If you cmd-f "buildRankings" in this diffs, it looks like it's used to just format the ranking data into a hash object when building a `GameRound` object